tests/functional/test_uninstall.py

```diff
@@ -166,20 +169,47 @@ def test_uninstall_entry_point(script):
         setup(
             name='ep-install',
             version='0.1',
-            entry_points={"pip_test.ep":
-                          ["ep:name1 = distutils_install",
-                           "ep:name2 = distutils_install"]
-                          }
+            entry_points={{"console_scripts": ["{0}", ],
+                           "pip_test.ep":
+                           ["ep:name1 = distutils_install",
+                            "ep:name2 = distutils_install"]
+                          }}
         )
-    """))
+    """.format(console_scripts)))
+    script_name = script.bin_path.join(console_scripts.split('=')[0].strip())
     result = script.pip('install', pkg_path)
+    assert script_name.exists
     result = script.pip('list', '--format=legacy')
     assert "ep-install (0.1)" in result.stdout
     script.pip('uninstall', 'ep_install', '-y')
+    assert not script_name.exists
     result2 = script.pip('list', '--format=legacy')
     assert "ep-install (0.1)" not in result2.stdout
 
 
+def test_uninstall_gui_scripts(script):
+    """
+    Make sure that uninstall removes gui scripts
+    """
+    pkg_name = "gui_pkg"
+    script.scratch_path.join(pkg_name).mkdir()
+    pkg_path = script.scratch_path / pkg_name
+    pkg_path.join("setup.py").write(textwrap.dedent("""
+        from setuptools import setup
+        setup(
+            name='{0}',
+            version='0.1',
+            entry_points={{"gui_scripts": ["test_ = distutils_install", ],
+                          }}
+        )
+    """.format(pkg_name)))
```

Comment:
With a recent rebase on master, this could be rewritten with `from tests.lib import create_test_package_with_setup
`, cf https://github.com/pypa/pip/blob/a12f2f493cc705f3acae104cf4dda5d0db519cf8/tests/functional/test_check.py#L25-L28